profiler/profiler.go

```diff
@@ -189,6 +195,30 @@ func (p *profiler) collect(ticker <-chan time.Time) {
 	}
 }
 
+// enabledProfileTypes returns the enabled profile types in a deterministic
+// order. The CPU profile always comes first because people might spot
```

Comment:
In my opinion, everything after the first sentence should go _inside_ the function, to make sure it's not missed.

Reply:
Depends. IMO it's important for the caller of this function to know what's going on, so having the relevant information showing up in the IDE when hovering over the call itself is useful IMO. I'm less worried about somebody modifying the function and not reading this comment. It seems unlikely that this function will see much editing going forward, and even less likely that somebody will miss the comment given that it's almost as big as the function itself : )

Reply:
As you wish. I'm not an advocate of optimising code for someone's editor or IDE.

Reply:
Function comments are a first-class feature of Go, so I think it's reasonable to assume that there are ergonomic ways for developers to view the comment associated with a function. Whether somebody uses godoc or their editor to view them should indeed be irrelevant, I just used "editor" as an example :)